File: core/src/main/java/com/codeabovelab/tpc/core/processor/SetAttributeAction.java
package com.codeabovelab.tpc.core.processor;

import com.codeabovelab.tpc.text.Text;

/**
 */
public class SetAttributeAction implements RuleAction<PredicateResult<?>> {
    private final String name;
    private final Object value;

    public SetAttributeAction(String name, Object value) {
        this.name = name;
        this.value = value;
    }

    @Override
    public void apply(ProcessingContext pc, Text text, PredicateResult<?> result) {
        pc.getAttributes().put(name, value);
    }
}

File: core/src/main/java/com/codeabovelab/tpc/core/processor/SetAttributeAction.kt
package com.codeabovelab.tpc.core.processor

import com.fasterxml.jackson.annotation.JsonTypeName

/**
 */
@JsonTypeName("SetAttributeAction")
class SetAttributeAction(
    val name: String,
    val value: Any
) : RuleAction<PredicateResult<*>> {

    override fun apply(context: RuleContext, predicateResult: PredicateResult<*>) {
        context.processingContext.attributes.put(name, value)
    }
}
